File: 2379-maximum-total-importance-of-roads/maximum-total-importance-of-roads.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        vector<long long> arr(n,0);
        for(auto it : roads){
            arr[it[0]]++;
            arr[it[1]]++;
        }
        sort(arr.begin(), arr.end());
        long long s = 0;
        for(int i = 0; i < n; i++){
            s += arr[i] * (i + 1);
        }
        return s;
    }
};

File: 2379-maximum-total-importance-of-roads/maximum-total-importance-of-roads.py
class Solution:
    def maximumImportance(self, n: int, roads: List[List[int]]) -> int:
        arr = [0] * n
        for a, b in roads:
            arr[a] += 1
            arr[b] += 1
        print(arr)
        arr.sort()
        print(arr)
        res = 0
        for i in range(n):
            res += arr[i] * (i + 1)
        return res